From 3488a253e7a20d491d180313340f39cf6fd043a9 Mon Sep 17 00:00:00 2001
From: warmcat <andy@warmcat.com>
Date: Fri, 25 Jul 2008 23:06:01 +0100
Subject: [PATCH] fix-gpio-led-patch-still-pwm-t3.patch

Starting up all the PWMs seems to be needed for PWM3 operation
and FIQ / HDQ / VIB operation.  But after starting, turn the
LEDs to GPIO-only.

Applies on top of Willie's patch

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/leds/leds-neo1973-gta02.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/leds/leds-neo1973-gta02.c b/drivers/leds/leds-neo1973-gta02.c
index e442e48..bf1d540 100644
--- a/drivers/leds/leds-neo1973-gta02.c
+++ b/drivers/leds/leds-neo1973-gta02.c
@@ -127,22 +127,22 @@ static int __init gta02led_probe(struct platform_device *pdev)
 
 		switch (lp->gpio) {
 		case S3C2410_GPB0:
-			lp->has_pwm = 0;
+			lp->has_pwm = 1;
 			lp->pwm.timerid = PWM0;
 			s3c2410_gpio_cfgpin(lp->gpio, S3C2410_GPB0_TOUT0);
 			break;
 		case S3C2410_GPB1:
-			lp->has_pwm = 0;
+			lp->has_pwm = 1;
 			lp->pwm.timerid = PWM1;
 			s3c2410_gpio_cfgpin(lp->gpio, S3C2410_GPB1_TOUT1);
 			break;
 		case S3C2410_GPB2:
-			lp->has_pwm = 0;
+			lp->has_pwm = 1;
 			lp->pwm.timerid = PWM2;
 			s3c2410_gpio_cfgpin(lp->gpio, S3C2410_GPB2_TOUT2);
 			break;
 		case S3C2410_GPB3:
-			lp->has_pwm = 0;
+			lp->has_pwm = 1;
 			lp->pwm.timerid = PWM3;
 			s3c2410_gpio_cfgpin(lp->gpio, S3C2410_GPB3_TOUT3);
 			break;
@@ -167,6 +167,7 @@ static int __init gta02led_probe(struct platform_device *pdev)
 			break;
 		default:
 			break;
+		}
 
 		mutex_init(&lp->mutex);
 		rc = led_classdev_register(&pdev->dev, &lp->cdev);
-- 
1.5.6.3

